src/libprojectM/MilkdropPreset/MilkdropShader.cpp

```diff
@@ -47,8 +49,9 @@ MilkdropShader::MilkdropShader(ShaderType type)
     } while (index < sizeof(m_randTranslation) / sizeof(m_randTranslation[0]));
 }
 
-void MilkdropShader::LoadCode(const std::string& presetShaderCode)
+void MilkdropShader::LoadCode(const std::string& presetShaderCode, bool dumpShaderCode)
 {
+    m_dumpShaderCode = dumpShaderCode;
```

Comment:
Debug stuff should better be in #ifdefs, so it doesn't affect release builds, potentially writing lots of files on user hard drives. See the other debug outputs in MilkdropPreset. When we implement a debug API for 4.2, this could be handled via the API, returning the shaders etc. via callbacks for example.

Since libprojectM is a library, it should not write anything to the filesystem or even output text to stdout/stderr.

Reply:
Removed, this was a local commit not intended for upstream.